project-base/app/src/Model/Product/ProductRepository.php

```diff
@@ -166,13 +166,29 @@ public function getSellableBySearchTextQueryBuilder(
      * @param \Shopsys\FrameworkBundle\Model\Pricing\Group\PricingGroup $pricingGroup
      * @return \Doctrine\ORM\QueryBuilder
      */
-    public function getAllSellableQueryBuilder($domainId, PricingGroup $pricingGroup): QueryBuilder
+    public function getAllListableQueryBuilder($domainId, PricingGroup $pricingGroup)
```

Comment:
rather than rewriting the `getAllListableQueryBuilder` method and removing the `$this->addDomain($queryBuilder, $domainId);`, I would slightly modify the `getAllOfferedQueryBuilder` method to use `QueryBuilderExtender->addOrExtendJoin()`. See e.g. https://github.com/shopsys/shopsys/blob/15.0/packages/framework/src/Model/Product/Filter/PriceRangeRepository.php#L104 - the extender service should handle multiple joins of the same entity. We can discuss this if needed :wink: